agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/dsl/extension/AIAgentEdges.kt

```diff
@@ -199,6 +200,24 @@ public infix fun <IncomingOutput, IntermediateOutput, OutgoingInput>
         .transformed { it.content }
 }
 
+
+/**
+ * Creates an edge that filters assistant messages based on a custom condition and extracts their content.
+ *
+ * @param block A function that evaluates whether to accept an assistant message
+ */
+public infix fun <IncomingOutput, OutgoingInput>
+        AIAgentEdgeBuilderIntermediate<IncomingOutput, List<Message.Response>, OutgoingInput>.onMultipleAssistantMessages(
```

Comment:
Not sure any model can even do it, but just in case let's keep it

Reply:
Well, basically it depends on node, `PromptExecutor.execute` always return the list and most of the times it's the list of one element, but it's still qualifies as multiple.
`requestLLM(): Message.Response` just makes `requestMultiple().first()`